test/receiveAndDeleteMode.spec.ts

```diff
@@ -136,7 +136,9 @@ describe("ReceiveBatch from Queue/Subscription", function(): void {
 });
 
 describe("Streaming Receiver from Queue/Subscription", function(): void {
+  let errorFromErrorHandler: Error | undefined;
```

Comment:
Overall found the test code confusing/hard to follow at first glance
Can rename to "uncaughtErrorFromHandlers"

Reply:
Even better, I think the error handling code is to just make sure the tests are all working correctly.
Technically it never gets invoked.
We can define a common onError handler for tests and move the verification check ti .afterEach() block. 
This way the tests are much cleaner and we don't need to add much code.

@ramya-rao-a @HarshaNalluru

Reply:
Good point.
Updated to have a common error handler.

But I havent moved the check on whether the error was thrown or not to afterEach() block.
There are currently checks after this check and before afterEach().
In the event where the unexpected error is thrown, these checks will fail as well.
If the unexpected error check moves to the afterEach() block, then it will never be run as the above checks will fail

Reply:
At a time only one error will be caught. Although ideally sequence of error catching is ideal like you have mentioned. I still think it's cleaner to have it in afterEach(). Doesnt seem incorrect, can sync up offline later. Not a blocker (Although I'm trying to better understand how to best structure the tests in general!)

Reply:
The major driving point for me is the scenario where 6 months down the line a change in the SDK breaks the test. If the sequence is not right, that engineer might end up wasting half the day chasing the wrong errors.

Reply:
Also, now that the check is just 1 line `chai.assert.fail()`, I dont think we have the issue of things not being clean anymore